var/spack/repos/builtin/packages/podio/package.py

```diff
@@ -163,7 +163,7 @@ class Podio(CMakePackage):
     patch("python-tests.patch", when="@:0.14.0")
 
     depends_on("root@6.08.06: cxxstd=17", when="cxxstd=17")
-    depends_on("root@6.28.04:", when="+rntuple")
+    depends_on("root@6.28.04: +root7", when="+rntuple")
     depends_on("root@6.28:", when="@0.17:")
     for cxxstd in ("17", "20"):
         depends_on("root cxxstd={}".format(cxxstd), when="cxxstd={}".format(cxxstd))
```

Comment:
I still don't know how the conretizer treats these additional requirements, after it has seen the others above (e.g. `+root7, when="rntuple"`. Are these applied on top of them? Otherwise, I think we should just switch to picking `cxxstd` from `root` during cmake argument building and add a `conflict` for `root cxxstd={11,14}`.

Reply:
Each depends_on becomes a condition (fact) that the final set of packages has to satisfy, so if there are multiple conditions on different variants, they all apply independently.

Reply:
Ah very nice. Thanks for the clarification.